retrieval/target.go

```diff
@@ -267,6 +266,36 @@ func (t *Target) String() string {
 	return t.host()
 }
 
+// fingerprint returns an identifying hash for the target.
+func (t *Target) fingerprint() model.Fingerprint {
+	t.RLock()
+	defer t.RUnlock()
+
+	lset := t.Labels()
+
+	lset[model.MetricsPathLabel] = t.labels[model.MetricsPathLabel]
```

Comment:
I think we should include all labels. This right now would clash on snmp/blackbox as it's one target with varying params.